### ISSUE TITLE
AveragePooling not supported in CPPEvalV2

### ISSUE THREAD

**Issue by dmagee** [Microsoft/CNTK#1179]
I'm trying to take a model generated by CNTK.exe and use it in C++ code. However on calling CNTK::Function::LoadModel() I get the exception:

"Unsupported ComputationNode with OperationName='AveragePooling' found when loading legacy CNTK model"

I'd be more than happy to move over to the new format, but the wiki suggests it is not possible to get CNTK.exe to generate output in the new model format (I'm using CNTK 2.0.beta2.0+ compiled from source on Windows).

Any thoughts on how to continue?

Thanks

Derek

N.B. As an aside I couldn't actually work out how to print out the exception in c-code, resorting to a debugger to find the error. Tried all of the below and it always just went to (...):

```
	catch (std::wstring err){
		std::cerr << "CNTK::Function::LoadModel() exception: " << err << std::endl;
		exit(-1);
	}
	catch (std::string err){
		std::cerr << "CNTK::Function::LoadModel() exception: " << err << std::endl;
		exit(-1);
	}
	catch (const char* err){
		std::cerr << "CNTK::Function::LoadModel() exception: " << err << std::endl;
		exit(-1);
	}
       // This one won't compile! Error	1	error C2338: Unsupported ValueType
	catch (std::invalid_argument err){
		std::cerr << "CNTK::Function::LoadModel() IA exception: " << err << std::endl;
		exit(-1);
	}
	catch (...){

		std::cerr << "CNTK::Function::LoadModel() exception" << std::endl;
		exit(-1);
	}
```
What is the data type I need to actually print the message?

**Comment by cha-zhang** [comment]
Try the latest beta 6.0. The issue has been fixed.

**Comment by Arminea** [comment]
Hi @dmagee, did you figure out something with your problem? I've got the same error message when I tried to load model in python code.

`Value error: The convolution map should have at least as many axes as the shape of the input it operates on.`

**Comment by dmagee** [comment]
@arminia no. I was hoping if I left it a bit over Christmas it might get fixed next time I downloaded the source. I hadn't had a chance to go back and try. I'm guessing not if you're using the latest version.

**Comment by Arminea** [comment]
I updated my project on December 30. So it's not the latest version I guess. 
@frankseide helped me with most of my previous issues. Maybe he will know something :)

**Comment by Arminea** [comment]
@dmagee Anything new? :)

**Comment by dmagee** [comment]
@arminea I've not had a chance to have another try I'm afraid, so it may have been fixed, but I'd hope they might post here if so.

**Comment by Arminea** [comment]
@dmagee Yesterday I found this [https://github.com/microsoft/CNTK/wiki/CNTK-model-format](url). I assume you made your model with brainscript just like me. It seems that there are two versions of model format and they are not compatible :( I think we will have to make a new model with python or c++. I've made a new issue for this if you wanna follow. [https://github.com/Microsoft/CNTK/issues/1355](url)

**Comment by dmagee** [comment]
@arminea you are correct I made my model with brainscript, I knew there was two formats, but CPPEvalV2 _should_ support both (as CNTK.exe can currently output only the old format). Some (simple) models trained using brainscript certainly do work  - including the example in the CNTK distribution.

**Comment by cha-zhang** [comment]
We have opened a new bug for this. We will fix it soon. Thanks!

**Comment by Arminea** [comment]
@cha-zhang Is it possible to keep us updated about it? :)

**Comment by cha-zhang** [comment]
Yes, I'll post here when it's fixed.

**Comment by Arminea** [comment]
Thank you very much :)

**Comment by zhouwangzw** [comment]
The bug has been fixed in master. If you can build bits from master, you would be get unblocked now. If you directly use binary download package, you may have to wait for our next release, which could be end of next week or a little later.

**Comment by mahilleb-msft** [comment]
Closing; if you find the issue persist, please re-open or file a new one. Thanks!
